### PR TITLE
Fix BINARY_SUBSCR_GETITEM stats

### PR DESCRIPTION
**Pull request by Fidget-Spinner**

### CODE REVIEW

**Comment by Fidget-Spinner** [comment]
Skipping news as this has no user impact.

**Comment by markshannon** [comment]
CI seem wonky

**Comment by markshannon** [comment]
I'm not sure slightly wrong stats merit a backport, but if it does maybe just backport the line `CALL_STAT_INC(inlined_py_calls);`?

File: Python/ceval.c [hunk 1]
```diff
@@ -2242,14 +2242,14 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, _PyInterpreterFrame *frame, int
             PyCodeObject *code = (PyCodeObject *)getitem->func_code;
             size_t size = code->co_nlocalsplus + code->co_stacksize + FRAME_SPECIALS_SIZE;
```
Comment (markshannon) [inline Python/ceval.c]: Unused code

**Comment by Fidget-Spinner** [comment]
> I'm not sure slightly wrong stats merit a backport, but if it does maybe just the stats fixes, and not the change to frame pushing?

I'll remove the backport label then. I wanted to backport this so that we have less backport conflicts in the future.

I changed the frame pushing because I copied the code then learned (from you) that theres a cleaner way to do it. So this should prevent people from copying the less clean example in the future.

**Comment by markshannon** [comment]
Looks good.

**Pull request merged**